lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.h

```diff
@@ -505,6 +506,8 @@ class TypeSystemSwiftTypeRef : public TypeSystemSwift {
 
   /// All lldb::Type pointers produced by DWARFASTParser Swift go here.
   ThreadSafeDenseMap<const char *, lldb::TypeSP> m_swift_type_map;
+  swift_demangle::LRUCache<std::string, CompilerType> m_canonical_types_cache;
```

Comment:
I wonder if these specialized caches per function are scalable though, as we'd need one cache per function.

I suppose we could only create caches for functions we know are very hot.

Reply:
Yes, I'm going to collect more detailed logs to see how many times a function is called with the same mangled name so we don't cache functions called one or two times.